### ISSUE TITLE
Null reference error when pydantic dataclass wrapped class inherits from stdlib dataclass without adding new field

### ISSUE THREAD

**Issue by joelliusczar** [pydantic/pydantic#4565]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Hi all,

I am encountering an exception when I have a pydantic dataclass that inherits from a stdlib dataclass but has no additional fields. (I use the stdlib dataclass for internal usages and pydantic for validation.)
I don't know if my use case is dumb, regardless here is the error trace I encounter for the example code below.

```
Traceback (most recent call last):
  File "/Users/username/Documents/Code/py/pydantic_bug/pydantic_dc_bug.py", line 22, in <module>
    class ValidatedDerivedC(UnvalidatedDataclass):
  File "pydantic/dataclasses.py", line 200, in pydantic.dataclasses.dataclass.wrap
    # https://bugs.python.org/issue33453 for details.
AttributeError: 'NoneType' object has no attribute '__bases__'
```

This is not a showstopper for me.
I've discovered that using `@pydantic.dataclasses.dataclass` (no parentheses) instead of `@pydantic.dataclasses.dataclass()` avoids the error.
Adding a field to the derived class also avoids the error.

I am mostly adding this bug report for awareness.

If you need any additional help, I sorta understand why this is happening, but at the moment I'm not certain how to fix it without breaking other stuff.

Thank you for your work on this library. It is much appreciated!

### Example Code

```Python
import dataclasses

import pydantic

@dataclasses.dataclass
class UnvalidatedDataclass:
    a: int = 0

#okay
@pydantic.dataclasses.dataclass
class ValidatedDerivedA(UnvalidatedDataclass):
    ...

#also okay
@pydantic.dataclasses.dataclass()
class ValidatedDerivedB(UnvalidatedDataclass):
    b: int = 0


#AttributeError: 'NoneType' object has no attribute '__bases__'
@pydantic.dataclasses.dataclass()
class ValidatedDerivedC(UnvalidatedDataclass):
    ...
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.2
pydantic compiled: True
install path: /Users/username/Documents/Code/py/pydantic_bug/env/lib/python3.9/site-packages/pydantic
python version: 3.9.10 (main, Jan 15 2022, 11:40:36)  [Clang 13.0.0 (clang-1300.0.29.3)]
platform: macOS-12.3.1-arm64-arm-64bit
optional deps. installed: ['typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [X] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by dolfinus** [comment]
Should be fixed in #4576

**Comment by samuelcolvin** [comment]
Confirmed, also confirmed this is new in v1.10.

**Comment by samuelcolvin** [comment]
Closed via #4878
